### ISSUE TITLE
iohub.server.sendResponse: Incorrect MAX_PACKET_SIZE

### ISSUE THREAD

**Issue by papr** [psychopy/psychopy#4184]
We are working on [integrating Pupil Core into PsychoPy's ioHub](https://github.com/pupil-labs/psychopy/tree/pupil-labs-pupil-core-eye-tracker) but are currently receiving a series of similar error messages:
```py
=============================
Error trying to send data to experiment process:
max_pkt_sz: 32748
reply_data_sz: 29407
pkt_cnt: -1
packet index, byte index: -1-1
Traceback (most recent call last):
  File "/Users/papr/work/tmp/psychopy/psychopy/iohub/server.py", line 370, in sendResponse
    self.socket.sendto(reply_data, address)
  File "/Users/papr/.pyenv/versions/3.6.10/envs/psychopy/lib/python3.6/site-packages/gevent/_socketcommon.py", line 703, in sendto
    return self._sock.sendto(*args)
OSError: [Errno 40] Message too long
=============================
```
As you can see `max_pkt_sz >= reply_data_sz`, but `reply_data_sz` is still too large for a single-message transfer. Therefore, I am assuming that [`max_pkt_sz` is calculated incorrectly](https://github.com/pupil-labs/psychopy/blob/bac0f417c1ce37b53cd5afe65d84aee77aab8de1/psychopy/iohub/server.py#L354).

Issue reproduced on
- macOS 11.4 Big Sur, Python 3.9
- macOS 10.15.7 Catalina, Python 3.6

**Comment by isolver** [comment]
Looks like the max udp packet size on macOS is ~9000 bytes. Atleast for now, if you keep each response to below 9k bytes you should not run into the issue. 

Out of interest, what are you needing to send that is 29k bytes in one response? For best IPC performance, if possible you probably want to keep each response to within a single packet.  

I'm looking into an actual fix, but simply reducing the max macOS packet size seems to cause another issue on macOS: If it is a multipacket response with > 2 packets it hangs.... investigating .....

**Comment by papr** [comment]
> Out of interest, what are you needing to send that is 29k bytes in one response? 

We currently have a very high `device_timer.interval`, resulting in ~250 samples being added to the event buffer per `_poll()` call. Am I assuming correctly that the iohub server tries to batch-transfer these? We might be doing something very wrong if a single event has a size fo 29k bytes...

**Comment by isolver** [comment]
> We currently have a very high device_timer.interval, resulting in ~250 samples being added to the event buffer per _poll() call.

Thanks for the extra info. You probably want the polling interval to be <= your device sampling rate so that each poll gets 0 to a couple events / samples. So if the tracker is sampling at 250 hz, a default device_timer.interval of 0.004 sec (4 msec) would work well. 

Regardless, I'll let you know when I've fixed the macOS multipacket issue. Thanks again.

**Comment by isolver** [comment]
This should be fixed in #4185. Thanks for raising the issue.
